File: ruby/hero.rb
# frozen_string_literal: true

class Hero
  def initialize(name, mainhand, offhand)
    @name = name
    @mainhand = mainhand
    @offhand = offhand
  end

  def to_s
    "#{@name} (#{@mainhand}, #{@offhand})"
  end

  def lmb
    "#{@name} uses #{@mainhand}!"
  end

  def rmb
    "#{@name} uses #{@offhand}!"
  end
end

File: python/hero.py
class Hero():

    def __init__(self, name, mainhand, offhand):
        self.name = name
        self.mainhand = mainhand
        self.offhand = offhand

    def __str__(self):
        return f"{self.name} ({self.mainhand}, {self.offhand})"

    def lmb(self):
        return f"{self.name} uses {self.mainhand}!"

    def rmb(self):
        return f"{self.name} uses {self.offhand}!"
